traits/ctraits.pyx

```diff
@@ -1370,15 +1370,8 @@ cdef object getattr_delegate(cTrait trait, CHasTraits obj, object name):
         delegate_attr_name = trait.delegate_attr_name(trait, obj, name)
         tp = type(delegate)
 
-        try:
-            result = getattr(delegate, delegate_attr_name)
-            return result
-        except NotImplementedError as exc:
-            raise exc
-        except Exception as exc:
-            raise DelegationError("The '%.50s' object has no attribute "
```

Comment:
To corroborate this, I ran the traits test suite with coverage enabled, and it looks like the whole block from L1826 to L1838 is never exercised. I don't know if that means the code is dead, but I expect so.

![schermafbeelding 2016-08-23 om 05 11 18](https://cloud.githubusercontent.com/assets/537559/17890047/be71bb96-68f0-11e6-9b4e-cc687ffa4dd5.png)